plugins/net_plugin/net_plugin.cpp

```diff
@@ -2489,7 +2495,18 @@ namespace eosio {
             boost::asio::bind_executor( strand,
               [conn = shared_from_this(), socket=socket]( boost::system::error_code ec, std::size_t bytes_transferred ) {
                // may have closed connection and cleared pending_message_buffer
-               if( !conn->socket_is_open() || socket != conn->socket ) return;
+               if (!conn->socket->is_open() && conn->socket_is_open()) { // if socket_open then close not called
+                  peer_dlog( conn, "async_read socket not open, closing");
+                  conn->close();
+                  return;
+               }
+               if (socket != conn->socket ) { // different socket, conn must have created a new socket, make sure previous is closed
+                  peer_dlog( conn, "async_read diff socket closing");
+                  boost::system::error_code ec;
+                  socket->shutdown( tcp::socket::shutdown_both, ec );
+                  socket->close( ec );
```

Comment:
The docs say that the destructor calls `cancel()` but it does not specify what if any shutdown is performed. Granted I do think is rather paranoid as it is likely that the destructor does clean up the connection.

Reply:
it takes time to find an answer, here is what I found:
https://stackoverflow.com/questions/39813066/boost-asio-socket-destructor-closes-connection

however we need to understand why do they recommend this `shutfdown` to be called. maybe because of reading? but we are not concerned about this. So I'm not hard on that just wanted to save few lines, net plugin code is quite verbose

Reply:
another good solution if you have some time and inspiration is to add custom deleter to shared ptr. that way we can save many duplicate code.